From 9606bc61184c42c2d784c63c309e8c2cfc815d1b Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Wed, 4 Nov 2015 12:31:57 -0500
Subject: [PATCH 08/12] ARM: dts: k2e: add device binding for crypto
 accelerator

This patch enables K2E HW crypto accelerator by adding device binding
for it.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 arch/arm/boot/dts/k2e-netcp.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/boot/dts/k2e-netcp.dtsi b/arch/arm/boot/dts/k2e-netcp.dtsi
index 320cf4d..5e1a742 100644
--- a/arch/arm/boot/dts/k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/k2e-netcp.dtsi
@@ -535,3 +535,27 @@ netcpx: netcpx@2f00000 {
 		};
 	};
 };
+
+crypto@24080000 {
+	compatible = "ti,netcp-sa-crypto";
+	label = "crypto-gbe";
+	reg = <0x24080000 0x40000>;
+	clocks = <&clksa>;
+
+	ti,navigator-dmas = <&dma_gbe 26>, <&dma_gbe 18>;
+	ti,navigator-dma-names = "crypto-rx", "crypto-tx";
+
+	rx-channel = "crypto-rx";
+	rx-queue-depth = <256 64 0 0>;
+	rx-buffer-size = <1500 4096 0 0>;
+	rx-compl-queue = <536>;
+	rx-pool	= <512 12>;
+
+	tx-channel = "crypto-tx";
+	tx-queue-depth = <256>;
+	tx-completion-queue = <537>;
+	tx-pool = <512 12>;
+	tx-submit-queue = <914>;
+
+	sc-id = <0x7000 0x71ff>;
+};
-- 
1.9.1

